test/sqllogictest/tpch_create_index.slt

```diff
@@ -501,27 +501,24 @@ materialize.public.q05_primary_idx:
           implementation
-            %0:customer » %1:orders[#1]KAiif » %2:lineitem[#0]KA » %3:supplier[#0, #1]KK » %4:nation[#0]KA » %5:region[#0]KAef
-            %1:orders » %0:customer[#0]KA » %2:lineitem[#0]KA » %3:supplier[#0, #1]KK » %4:nation[#0]KA » %5:region[#0]KAef
-            %2:lineitem » %1:orders[#0]KAiif » %0:customer[#0]KA » %3:supplier[#0, #1]KK » %4:nation[#0]KA » %5:region[#0]KAef
-            %3:supplier » %0:customer[#3]KA » %1:orders[#1]KAiif » %2:lineitem[#0, #1]KK » %4:nation[#0]KA » %5:region[#0]KAef
-            %4:nation » %5:region[#0]KAef » %0:customer[#3]KA » %1:orders[#1]KAiif » %2:lineitem[#0]KA » %3:supplier[#0, #1]KK
-            %5:region » %4:nation[#2]KA » %0:customer[#3]KA » %1:orders[#1]KAiif » %2:lineitem[#0]KA » %3:supplier[#0, #1]KK
+            %0:customer » %1:orders[#1]KAiif » %2:lineitem[#0]KA » %3:supplier[#3]KA » %4:nation[#0]KA » %5:region[#0]KAef
+            %1:orders » %0:customer[#0]KA » %2:lineitem[#0]KA » %3:supplier[#3]KA » %4:nation[#0]KA » %5:region[#0]KAef
+            %2:lineitem » %1:orders[#0]KAiif » %0:customer[#0]KA » %3:supplier[#3]KA » %4:nation[#0]KA » %5:region[#0]KAef
+            %3:supplier » %0:customer[#3]KA » %1:orders[#1]KAiif » %2:lineitem[#2]KA » %4:nation[#0]KA » %5:region[#0]KAef
+            %4:nation » %5:region[#0]KAef » %0:customer[#3]KA » %1:orders[#1]KAiif » %2:lineitem[#0]KA » %3:supplier[#3]KA
+            %5:region » %4:nation[#2]KA » %0:customer[#3]KA » %1:orders[#1]KAiif » %2:lineitem[#0]KA » %3:supplier[#3]KA
           ArrangeBy keys=[[#0{c_custkey}], [#3{c_nationkey}]] // { arity: 8 }
             ReadIndex on=customer pk_customer_custkey=[delta join 1st input (full scan)] fk_customer_nationkey=[delta join lookup] // { arity: 8 }
           ArrangeBy keys=[[#0{o_orderkey}], [#1{o_custkey}]] // { arity: 9 }
             ReadIndex on=orders pk_orders_orderkey=[delta join lookup] fk_orders_custkey=[delta join lookup] // { arity: 9 }
-          ArrangeBy keys=[[#0{l_orderkey}], [#0{l_orderkey}, #1{l_suppkey}]] // { arity: 4 }
-            Project (#0{l_orderkey}, #2{l_suppkey}, #5{l_extendedprice}, #6{l_discount}) // { arity: 4 }
-              ReadIndex on=lineitem pk_lineitem_orderkey_linenumber=[*** full scan ***] // { arity: 16 }
-          ArrangeBy keys=[[#0{s_suppkey}, #1{s_nationkey}]] // { arity: 2 }
-            Project (#0{s_suppkey}, #3{s_nationkey}) // { arity: 2 }
-              Filter (#0{s_suppkey}) IS NOT NULL // { arity: 7 }
-                ReadIndex on=supplier pk_supplier_suppkey=[*** full scan ***] // { arity: 7 }
+          ArrangeBy keys=[[#0{l_orderkey}], [#2{l_suppkey}]] // { arity: 16 }
+            ReadIndex on=lineitem fk_lineitem_orderkey=[delta join lookup] fk_lineitem_suppkey=[delta join lookup] // { arity: 16 }
+          ArrangeBy keys=[[#3{s_nationkey}]] // { arity: 7 }
+            ReadIndex on=supplier fk_supplier_nationkey=[delta join lookup] // { arity: 7 }
```

Comment:
Good: 3 less arrangements. (This is the query that originally prompted this investigation.)